### ISSUE TITLE
PHP 8.2 support?

### ISSUE THREAD

**Issue by sosherof** [Bandwidth/php-bandwidth-iris#95]
Does this library support PHP 8.2?  Has it been given similar attention as bandwidth/php-sdk regarding type hinting?

**Comment by sosherof** [comment]
Funny enough, my reason for wanting to use your libraries was to avoid the XML as well.   Is there any thought on API v3 that uses JSON instead?

**Comment by ajrice6713** [comment]
No one likes XML - we are working internally to rev this dashboard/numbers API to a new major version that supports JSON, however I dont have an official timeline on that.

**Comment by sosherof** [comment]
While I'm here making suggestions and especially if you're considering an API v3, consider collapsing the order status APIs down into one.  Right now you must use a completely different endpoint to that check that order status for just about every API call that produces an OrderId.  For example, if I import a TN and then want to assign a campaign, I have to wait on the import to COMPLETE.  The import order status URL is different from the TnOptionsOrder status URL, etc.

**Comment by kkamath-bandwidth** [comment]
Re: json, its a yes, but not right now. So, current goal for v2 is to internationalize (e164 support & 60+ countries) with minimum code change on behalf of our existing customers. I promise to push for JSON support after we finish that development (just trying to not hold unification hostage to this and since XML needs to be supported on old endpoints anyways). Also, the new endpoints we are adding will have JSON, so we are learning incrementally too!

Re: order status, its a maybe Sam. The order is not an ecommerce like order that was intended to be a common thing, but instead an async pattern. Bur, yes I am sure there is scope for improvement here. We can look into this more during JSON support,

**Comment by ajrice6713** [comment]
@sosherof i know its been a while - I just updated our tests to run on 8.0 -> 8.3 and everything seems to be passing if you would like to check v4.2.2 and confirm!

**Comment by sosherof** [comment]
Is v4.2.2 the current release?  README says 3.3.2.  Regardless, if the minimum supported version of PHP is now 8.0, it'll be a moment before I can test as I'll need to setup a test site to run only on the dev server (prod server is still running PHP 7.2).

**Comment by ajrice6713** [comment]
Yes - I can get that readme updated - 4.2.2 is the current version - you can see our releases [here](https://github.com/Bandwidth/php-bandwidth-iris/releases).
